src/glade.py

```diff
@@ -74,18 +93,17 @@ def __repr__(self): return repr(self.o) if self.o else ''
 # In either case, P alpha Q is ranked last
 # Note that candidate repetitions and candidate alternations can
 # be ordered independently
+# We don't genralize all decendandts in one go, but only one substring at each step. Section 4.1 page 4
```

Comment:
Can you add the relevant text here (keep the Section 4.1, page 4, but add the relevant text if there is one).